### PR TITLE
Editing in Mapsui.Nts instead of Mapsui.Samples.Wpf.Editing

### PR DESCRIPTION
**Pull request by inforithmics**

### CODE REVIEW

**Comment by pauldendulk** [comment]
Interesting. Tell me more! What is the plan? At the time Mapsui.Samples.Wpf.Editing was created there was a lot of specific code that I wanted to have in Mapsui but was not ready yet. The dependency on WPF was still too big. Is this something you are working on?

**Comment by inforithmics** [comment]
Actually it isn't a lot wpf specific code anymore. In EditManipulation it is Keyboard.IsKeyDown and MouseMove and MouseLeftButtonDown and MouseLeftButtonUp evemts

So I think I plan to introduce an IMapControlEdit interface which has this Method and Events in a platform neutral way. So when a Platform implements IMapControlEdit editing works. Currently all platforms display the edit sample but it doesn't work yet.

**Comment by inforithmics** [comment]
Editing Sample works now with the Wpf MapControl in the others it displays

**Comment by inforithmics** [comment]
Things done:
Editing in Blazor

**Comment by pauldendulk** [comment]
> IMapControlEdit

This does not sound attractive to me (although some changes the MapControl may be necessary). Instead I would prefer components which could be added without special a implementation needed in the MapControl. Also perhaps Widgets could be used. If we could use SkiaSharp edit buttons it could work on all platforms without any platform specific code.

File: Mapsui.UI.Blazor/MapControl.cs [hunk 1]
```diff
@@ -163,11 +163,44 @@ private protected void RunOnUIThread(Action action)
         action();
     }
 
+    [SuppressMessage("Usage", "VSTHRD100:Avoid async void methods")]
+    protected async void OnDblClick(MouseEventArgs e)
```
[the remark this answers is not in the document]
Comment (pauldendulk) [reply Mapsui.UI.Blazor/MapControl.cs:167]: On the other hand. The fact that we can show editing in this stage already in our Blazor samples is absolutely wonderful! So, for the current stage we're in this is actually a great improvement, although I think it should be considered temporary.

**Review by pauldendulk — APPROVED** [review]
Great improvement

**Comment by inforithmics** [comment]
You are right I looked at the widget implementation, And I think with adding

public bool HandleWidgetMove(Navigator navigator, MPoint position) i could handle the Editing in a Edit Surface Widget. The Edit Buttons are simply handled with the HandleWidgetTouched. What I don't know yet is if I need to add an HandleWidgetTouching. ( mouseDown, the HandleWidgetTouched is mouseUp)

**Comment by inforithmics** [comment]
The widget editing changes, I will add them in a separate pull request.

**Pull request merged**